### PR TITLE
refactor: use mockHttpclient instead of mockUrllib

### PR DESCRIPTION
**Pull request by fengmk2**
closes https://github.com/eggjs/egg/issues/243

##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: app/extend/application.js [hunk 1]
```diff
@@ -321,10 +321,15 @@ module.exports = {
         }
         return mockResult;
       }
-      return yield requestThunk.call(urllib, url, opt);
+      return yield rawRequest.call(httpclient, url, opt);
     }
   },
 
+  mockUrllib(...args) {
```
Comment (fengmk2) [inline app/extend/application.js]: 加个测试用例

File: app/extend/application.js [hunk 2]
```diff
@@ -321,10 +321,15 @@ module.exports = {
...
+    this.deprecate('[egg-mock] Please use app.mockHttpclient instead of app.mockUrllib');
+    return this.mockHttpclient(...args);
```
Comment (atian25) [inline app/extend/application.js]: 完全移除对6的支持了么？那其他单元测试的 ci 配置都要改改

**Comment by popomore** [comment]
这个应该要发 major 了

File: app/extend/application.js (hunk 2 above)
Comment (atian25) [inline app/extend/application.js]: egg-ci 的默认值也要改改

**Comment by fengmk2** [comment]
我发个 major 吧

File: app/extend/application.js (hunk 2 above)
Comment (fengmk2) [inline app/extend/application.js]: 好

**Comment by fengmk2** [comment]
2.0.0, 1.3.1 是原来的 1.2.1，用于覆盖 1.3.0

```
+ egg-mock@2.0.0
latest: 1.3.0
release-1: 1.3.1
```

**Pull request merged**